theme/scripts/logo.ts

```diff
@@ -7,19 +9,39 @@ export default (function() {
   const Logo = {
     SELECTORS: {
       logoText: "[data-logo-text]",
+      logoTextMobile: "[data-logo-text-mobile]",
     },
     CONSTANTS: {
       index: 'index'
     },
+    STATE: {
+      setIntervalId: null
+    },
 
     init() {
       const logoEl: HTMLElement = document.querySelector(Logo.SELECTORS.logoText);
-      Logo.bindEventListeners(logoEl);
+      const logoElMobile: HTMLElement = document.querySelector(Logo.SELECTORS.logoTextMobile);
+      Logo.bindEventListeners(logoEl, logoElMobile);
     },
 
-    bindEventListeners(logoEl: HTMLElement) {
+    bindEventListeners(logoEl: HTMLElement, logoElMobile: HTMLElement) {
       logoEl.addEventListener('mouseenter', () => Logo.randomizeLogo(logoEl));
-      logoEl.addEventListener('mouseleave', () => Logo.resetLogo(logoEl))
+      logoEl.addEventListener('mouseleave', () => Logo.resetLogo(logoEl));
+
+      window.addEventListener('DOMContentLoaded', () => Logo.checkBreakpointAndRandomizeLogo(logoElMobile));
+      window.addEventListener('resize', _debounce(() => 
+      Logo.checkBreakpointAndRandomizeLogo(logoElMobile), 150));
+    },
+
+    checkBreakpointAndRandomizeLogo(logoElMobile: HTMLElement) {
+      if (isMobile()) {        
```

Comment:
Hey, so I'm still noticing an issues where `randomizeLogo` seems to be getting called much more frequently than every two seconds. You can recreate it but resizing the browser slowly to a small breakpoint. 

After  checking `if(isMobile()` we'll also want to check `if (Logo.STATE.setIntervalId)` otherwise, everytime `isMobile()` evaulates to true, we are adding another setInterval, and the logo starts randomizing double time. 

In the case that `isMobile` and an existing setInterval has already been set, we should just return and do nothing.

Reply:
Wow I don't know how I missed that check for `if (Logo.STATE.setIntervalId)` from your previous notes, thank you for the reminder!! 🤯 So do you think that was the main issue causing `randomizeLogo` to fire too many times (+ me using `throttle` instead of `debounce`)? 

After adding the check for `if (Logo.STATE.setIntervalId)`, it doesn't seem to be firing too many times: https://share.getcloudapp.com/Blu4OGnv

Reply:
I honestly don't think it was throttle at all, if you update it to use throttle now, it should still work, but debouncing is the right call here, since we only want to call this listener once the resize event has finished. 

Yeah, I'm pretty sure the reason we were seeing that bugged out behavior is because `isMobile()` was evaluating to true everytime the resize event triggered, which was creating another setInterval, and without clearing the previous one.

Reply:
Just tested throttle and you're right!! Thanks again : )